x-pack/plugins/lens/public/visualizations/datatable/components/table_basic.tsx

```diff
@@ -271,24 +272,20 @@ export const DatatableComponent = (props: DatatableRenderProps) => {
       firstLocalTable.columns.reduce<Record<string, boolean>>(
         (map, column) => ({
           ...map,
-          [column.id]: column.meta.type === 'number',
+          [column.id]: isNumericField(column.meta),
         }),
         {}
       ),
-    [firstLocalTable]
+    [firstLocalTable.columns]
   );
 
   const alignments: Record<string, 'left' | 'right' | 'center'> = useMemo(() => {
     const alignmentMap: Record<string, 'left' | 'right' | 'center'> = {};
     columnConfig.columns.forEach((column) => {
-      if (column.alignment) {
-        alignmentMap[column.columnId] = column.alignment;
-      } else {
-        alignmentMap[column.columnId] = isNumericMap[column.columnId] ? 'right' : 'left';
-      }
+      alignmentMap[column.columnId] = getColumnAlignment(column, isNumericMap[column.columnId]);
```

Comment:
nit: do you mind updating these two objects (alignmentMap and isNumericMap) to a proper `Map` type?
This because we could always have problems with manually configured `columnId`s and JS prototype pollution, is always better to use Map/Set when the data is not fully under our control (aka a saved object can be modified and imported)

Reply:
Good call, fixed in 7ca5ed2 along with changing `minMaxByColumnId`.

Reply:
pefect thanks